core/src/test/scala/integration/kafka/api/BaseQuotaTest.scala

```diff
@@ -136,6 +141,28 @@ abstract class BaseQuotaTest extends IntegrationTestHarness {
     assertTrue("Should have been throttled", consumerThrottleMetric.value > 0)
   }
 
+  @Test
+  def testThrottledRequest() {
+
+    overrideQuotas(Long.MaxValue, Long.MaxValue, 0.1)
+    waitForQuotaUpdate(Long.MaxValue, Long.MaxValue, 0.1)
+
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singleton(topic1))
+    val endTimeMs = System.currentTimeMillis + 10000
+    var throttled = false
+    while (!throttled && System.currentTimeMillis < endTimeMs) {
+      consumer.poll(100)
```

Comment:
Consumers in this test are configured with `fetch.max.wait.ms=0`. The quota is set very small as well to trigger throttling.